include/gdsync/core.h

```diff
@@ -166,10 +161,10 @@ int gds_stream_post_send_all(CUstream stream, int count, gds_send_request_t *req
  * Represents a wait operation on a particular CQ
  */
 
-typedef struct gds_wait_request {
-        struct ibv_exp_peer_peek peek;
-        struct peer_op_wr wr[GDS_WAIT_INFO_MAX_OPS];
-} gds_wait_request_t;
+typedef struct gds_wait_request gds_wait_request_t;
+
+int gds_alloc_wait_request(gds_wait_request_t **request, int num);
```

Comment:
I'd rather not introduce ctors for request objects.
The usual trick is to define for example:
```
   typedef struct gds_wait_request *gds_wait_request_t;
```
And then to automatically allocate the objects internally to the APIs.

Reply:
Some people recommend against hiding a pointer type behind typedef. One reason is that it can break expectation of "const". See https://wiki.sei.cmu.edu/confluence/display/c/DCL05-C.+Use+typedefs+of+non-pointer+types+only for more details.

Concerning ctors, what will you recommend instead?

Reply:
I still don't think we should have explicit ctor/dtor APIs.
Are they really needed?

Reply:
No, we don't. I forgot to remove these functions. Will fix in the next revision.